### PR TITLE
Update RCTScrollView.m

### PR DESCRIPTION
**Pull request by Feng999**
fix problem of function scrollToEnd: There are some strange thing happened when contentSize.height(width)  is smaller than bounds.size.height(width).  In fact, there is no need to scroll in this case.

Thanks for submitting a PR! Please read these instructions carefully:

- [x] Explain the **motivation** for making this change.
- [x] Provide a **test plan** demonstrating that the code is solid.
- [x] Match the **code formatting** of the rest of the codebase.
- [x] Target the `master` branch, NOT a "stable" branch.

## Motivation (required)

What existing problem does the pull request solve?

## Test Plan (required)

A good test plan has the exact commands you ran and their output, provides screenshots or videos if the pull request changes UI or updates the website. See [What is a Test Plan?][1] to learn more.  

If you have added code that should be tested, add tests.

## Next Steps

Sign the [CLA][2], if you haven't already.

Small pull requests are much easier to review and more likely to get merged. Make sure the PR does only one thing, otherwise please split it.

Make sure all **tests pass** on both [Travis][3] and [Circle CI][4]. PRs that break tests are unlikely to be merged.

For more info, see the ["Pull Requests"][5] section of our "Contributing" guidelines.

[1]: https://medium.com/@martinkonicek/what-is-a-test-plan-8bfc840ec171#.y9lcuqqi9
[2]: https://code.facebook.com/cla
[3]: https://travis-ci.org/facebook/react-native
[4]: http://circleci.com/gh/facebook/react-native
[5]: https://github.com/facebook/react-native/blob/master/CONTRIBUTING.md#pull-requests

### CODE REVIEW

**Comment by Guardiannw** [comment]
This is about the same that I have in my PR #12889.  Definitely need one of these implemented.

**Comment by Feng999** [comment]
Nothing changed to android code,  don't know why test failed.

File: React/Views/RCTScrollView.m [hunk 1]
```diff
@@ -529,9 +529,13 @@ - (void)scrollToEnd:(BOOL)animated
   BOOL isHorizontal = [self isHorizontal:_scrollView];
   CGPoint offset;
   if (isHorizontal) {
-    offset = CGPointMake(_scrollView.contentSize.width - _scrollView.bounds.size.width, 0);
+    CGFloat offsetX = _scrollView.contentSize.width - _scrollView.bounds.size.width;
+    offsetX = offsetX > 0 ? offsetX : 0;
+    offset = CGPointMake(offsetX, 0);
```
Comment (javache) [inline React/Views/RCTScrollView.m]: Can you do `CGPointMake(MAX(offsetX, 0), 0))` instead?

File: React/Views/RCTScrollView.m [hunk 2]
```diff
@@ -529,9 +529,13 @@ - (void)scrollToEnd:(BOOL)animated
...
   } else {
-    offset = CGPointMake(0, _scrollView.contentSize.height - _scrollView.bounds.size.height);
+    CGFloat offsetY = _scrollView.contentSize.height - _scrollView.bounds.size.height;
+    offsetY = offsetY > 0 ? offsetY : 0;
+    offset = CGPointMake(0, offsetY);
```
Comment (javache) [inline React/Views/RCTScrollView.m]: Can you do `CGPointMake(0, MAX(offsetY, 0))` instead?

Comment (Feng999) [inline React/Views/RCTScrollView.m]: Good advice！ Thank you.

File: React/Views/RCTScrollView.m [hunk 3]
```diff
@@ -525,9 +525,11 @@ - (void)scrollToEnd:(BOOL)animated
   BOOL isHorizontal = [self isHorizontal:_scrollView];
   CGPoint offset;
   if (isHorizontal) {
-    offset = CGPointMake(_scrollView.contentSize.width - _scrollView.bounds.size.width, 0);
+    CGFloat offsetX = _scrollView.contentSize.width - _scrollView.bounds.size.width;
+    offset = CGPointMake(MAX(offsetX, 0), 0);
   } else {
-    offset = CGPointMake(0, _scrollView.contentSize.height - _scrollView.bounds.size.height);
+    CGFloat offsetY = _scrollView.contentSize.height - _scrollView.bounds.size.height;
```
Comment (shergin) [inline React/Views/RCTScrollView.m]: I am afraid that we forget to consider `contentInset` here. :(

**Pull request closed without merging**